plugins/kubernetes-backend/src/service/KubernetesErrorIndicator.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Logger } from 'winston';
+import {
+  KubernetesFetchError,
+  KubernetesErrorTypes,
+} from '@backstage/plugin-kubernetes-common';
+import { KubernetesRejectionHandler } from '../types/types';
+import { HttpError } from '@kubernetes/client-node';
+
+const knownStatusCodeErrors: Record<number, KubernetesErrorTypes> = {
+  400: 'BAD_REQUEST',
+  401: 'UNAUTHORIZED_ERROR',
+  404: 'NOT_FOUND',
+  500: 'SYSTEM_ERROR',
+};
+
+export class KubernetesErrorIndicator implements KubernetesRejectionHandler {
+  private readonly logger: Logger;
+  private readonly statusCodeErrorTypes: Record<number, KubernetesErrorTypes>;
+  private readonly signature: string = 'KubernetesFetchError';
+  constructor(
+    logger: Logger,
+    statusCodeErrorTypes: Record<
+      number,
+      KubernetesErrorTypes
+    > = knownStatusCodeErrors,
+  ) {
+    this.logger = logger;
+    this.statusCodeErrorTypes = statusCodeErrorTypes;
+  }
+  public onRejected: (
+    reason: any,
+    resourcePath?: string,
+  ) => Promise<KubernetesFetchError> = (reason, resourcePath) => {
```

Comment:
this shouldn't be an arrow function, just make it a proper member like the other methods